hles/extraction/src/test/scala/org/broadinstitute/monster/dap/MockOkWrapper.scala

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.monster.dap
+
+import okhttp3._
+import upack.Msg
+
+import scala.concurrent.{Future, Promise}
+
+class MockOkWrapper(
+  responseMap: Map[RedcapRequestGeneratorParams, Msg] = Map.empty
```

Comment:
🤔 Would it be simpler to just take a list of responses and spit them out in order they were provided, rather than having to finding them in the mapping and having to do the object equality checks?

Reply:
i'm hesitant to regurgitate responses without verifying that we're responding to the expected request, since it could lead to hard to track down bugs or, worse, tests that pass despite the introduction of a request that will generate different data from the live API.